src/compiler/commandLineParser.ts

```diff
@@ -723,10 +727,22 @@ namespace ts {
                 const supportedExtensions = getSupportedExtensions(options);
                 Debug.assert(indexOf(supportedExtensions, ".ts") < indexOf(supportedExtensions, ".d.ts"), "Changed priority of extensions to pick");
 
+                const potentialFiles: string[] = [];
+                if (host.readDirectoryWithMultipleExtensions) {
+                    addRange(potentialFiles, host.readDirectoryWithMultipleExtensions(basePath, supportedExtensions, exclude));

```

Comment:
I believe this already handled by @riknoll in https://github.com/Microsoft/TypeScript/pull/8841. can we make sure we are doing one approach for both PRs.